pkg/resource/deploy/step_generator.go

```diff
@@ -853,6 +853,103 @@ func processIgnoreChanges(inputs, oldInputs resource.PropertyMap,
 	return ignoredInputs.ObjectValue(), nil
 }
 
+func copyWithUnknownElements(oldValue resource.PropertyValue) resource.PropertyValue {
+	unknown := resource.MakeComputed(resource.NewStringProperty(""))
+	switch {
+	case oldValue.IsObject():
+		newValue := resource.PropertyMap{}
+		for k := range oldValue.ObjectValue() {
+			newValue[k] = unknown
+		}
+		return resource.NewObjectProperty(newValue)
+	case oldValue.IsArray():
+		newValue := make([]resource.PropertyValue, len(oldValue.ArrayValue()))
+		for i := range newValue {
+			newValue[i] = unknown
+		}
+		return resource.NewArrayProperty(newValue)
+	default:
+		return unknown
+	}
+}
+
+func copyStableValue(oldState, newState resource.PropertyMap, path resource.PropertyPath, allowPartialValues bool) {
+	if len(path) == 0 || len(path) > 1 && !allowPartialValues {
```

Comment:
i'm honestly dense about precedence (too much time writing parsers).  so parenthesizing this would be helpful.